Sources/Datadog/RUM/RUMMonitor/Scopes/RUMSessionScope.swift

```diff
@@ -14,27 +14,59 @@ internal class RUMSessionScope: RUMScope {
         static let sessionMaxDuration: TimeInterval = 4 * 60 * 60 // 4 hours
     }
 
+    // MARK: - Child Scopes
+
+    /// Active View scopes. Scopes are added / removed when the View starts / stops displaying.
+    private(set) var viewScopes: [RUMScope] = []
+
     // MARK: - Initialization
 
     unowned let parent: RUMScope
     private let dependencies: RUMScopeDependencies
 
-    /// This session UUID.
-    private var sessionUUID: UUID
-    /// The start time of this session.
+    /// This Session UUID.
+    private var sessionUUID: RUMUUID
+    /// The start time of this Session.
     private var sessionStartTime: Date
-    /// Time of the last RUM interaction noticed by this session.
+    /// Time of the last RUM interaction noticed by this Session.
     private var lastInteractionTime: Date
 
     init(
         parent: RUMScope,
-        dependencies: RUMScopeDependencies
+        dependencies: RUMScopeDependencies,
+        startTime: Date
     ) {
         self.parent = parent
         self.dependencies = dependencies
-        self.sessionUUID = UUID()
-        self.sessionStartTime = dependencies.dateProvider.currentDate()
-        self.lastInteractionTime = self.sessionStartTime
+        self.sessionUUID = dependencies.rumUUIDGenerator.generateUnique()
+        self.sessionStartTime = startTime
+        self.lastInteractionTime = startTime
+    }
+
+    /// Creates a new Session upon expiration of the previous one.
+    convenience init(
+        from expiredSession: RUMSessionScope,
+        startTime: Date
+    ) {
+        self.init(
+            parent: expiredSession.parent,
+            dependencies: expiredSession.dependencies,
+            startTime: startTime
+        )
+
+        // Transfer active Views by creating new `RUMViewScopes` for their identity objects:
+        self.viewScopes = expiredSession.viewScopes.compactMap { expiredView in
+            guard let expiredView = expiredView as? RUMViewScope, let expiredViewIdentity = expiredView.identity else {
```

Comment:
maybe we should check `(identity as UIViewController).window == nil` here? 
sometimes apps keep VCs alive although they are not part of view hierarchy, do we consider this case as "view is alive" or not?

Reply:
Good idea 👍. I think it's best to have both: check if the object exists + check if it's removed from the window. This way we could be even sending "reference leak" RUM Erorrs in such case 🙌.

Reply:
Done in https://github.com/DataDog/dd-sdk-ios/pull/193/commits/50693ac26132de67b675929125259bafff461b23. I've also created `RUMM-634 Produce a RUM error when the UIViewController is leaked`.